07-Extract-NDVI-From-Rasters-in-R.Rmd

```diff
@@ -161,104 +166,122 @@ names(avg_NDVI_HARV) <- "meanNDVI"
 avg_NDVI_HARV$site <- "HARV"
 
-```
+#add a "year" column to our data
+avg_NDVI_HARV$year <- "2011"
 
+#view data
+head(avg_NDVI_HARV)
+```
 
+We now have data frame with each file and two values for each one `meanNDVI` and 
+`site`.  
 
 #Extract Julian Day from row.names
+We'd like to produce a plot where Julian days (the numeric day of the year,
+0 - 365/366) is on the x-axis and NDVI is on
+the y-axis. To do this, we'll need a column that is populated with only the
+Julian day value for each NDVI value. We can do this using one of several
+approaches. 
 
-We'd like to produce a plot where Julian Days is on the X axis and NDVI is on the 
-Y axis. To do this, we'll need a column that is populated with Julian Day values
-for each NDVI value. We can do this using several approaches. For this lesson, 
-e will use `gsub` to replace the `X` and the `_HARV_ndvi_crop`.
+For this lesson, we will use `gsub` to replace the `X` and the `_HARV_ndvi_crop`.
 
 ```{r extract-julian-day }
-
-#note the use of | is equivalent to "or". this allows us to search for more than
-#one pattern in our text strings
+#note the use of the vertical bar character ( | ) is equivalent to "or". This
+# allows us to search for more than one pattern in our text strings.
 julianDays <- gsub(pattern = "X|_HARV_ndvi_crop", #the pattern to find 
             x = row.names(avg_NDVI_HARV), #the object containing the strings
             replacement = "") #what to replace each instance of the pattern with
 
+#alternate format
+#julianDays <- gsub("X|_HARV_ndvi_crop", "", row.names(avg_NDVI_HARV))
+
 #make sure output looks ok
 head(julianDays)
 
-#add julianDay values as a column in the dataframe
+#add julianDay values as a column in the data frame
 avg_NDVI_HARV$julianDay <- julianDays
 
 #what class is the new column
 class(avg_NDVI_HARV$julianDay)
-
-#convert julian days into a time class
-#avg_NDVI_HARV$julianDay <- as.POSIXlt(avg_NDVI_HARV$julianDay, format="%j")
-
-#the code above does weird things. for one i think jday is 0 based indexing
-#but also it seems to want a year and defaults to 2015. i am not sure how to force 
-#it to 2011 which is when the data were collected
-#being lazy and useing int for the time being.
-avg_NDVI_HARV$julianDay <- as.integer(avg_NDVI_HARV$julianDay)
-
-# Parse that character vector
-#strptime(date_info, "%Y %j")
 ```
 
-#Scale Data
-The last thing that we need to do it so scale our data. Valid NDVI values range 
-between 0-1. The metadata included a `ScaleFactor`. NDVI data are scaled by a 
-factor of 10,000. A scale factor is commonly used in larger datasets. Storing decimal
-places actually consumes more space and creates larger file sizes compared to 
-storing integer values.
-
-```{r scale-data }
 
-#scale data by 10,000
-avg_NDVI_HARV$meanNDVI <- avg_NDVI_HARV$meanNDVI / 10000
-#view output
-head(avg_NDVI_HARV)
+##Create Date Column from Julian Day
+Currently the Julian day is recognized as a character class value.  We'd prefer
```

Comment:
A date is more useful for plotting and analysis, when it is stored as an `R` date class.